sections/futures/ProfitCalculator/ProfitCalculator.tsx

```diff
@@ -0,0 +1,251 @@
+import { useEffect, useState } from 'react';
+import { ethers } from 'ethers';
+import { BigNumber } from '@ethersproject/bignumber';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+
+import PnLs from './PnLs';
+import ProfitDetails from './ProfitDetails';
+import BaseModal from 'components/BaseModal';
+import LabelWithInput from './LabelWithInput';
+import PositionButtons from '../../../sections/futures/PositionButtons';
+import { PositionSide } from '../types';
+
+const scalar = 100;
+
+const ProfitCalculator = ({ marketAsset, marketAssetRate, setOpenProfitCalcModal }: any) => {
+	const { t } = useTranslation();
+
+	// BigNumbers
+	const [entryPrice, setEntryPrice] = useState<BigNumber>(ethers.BigNumber.from(0));
+	const [exitPrice, setExitPrice] = useState<BigNumber>(ethers.BigNumber.from(0));
+	const [stopLoss, setStopLoss] = useState<BigNumber>(ethers.BigNumber.from(0));
+	const [marketAssetPositionSize, setMarketAssetPositionSize] = useState<BigNumber>(
+		ethers.BigNumber.from(0)
+	);
+	// Custom type
+	const [leverageSide, setLeverageSide] = useState<PositionSide>(PositionSide.LONG);
+
+	const handleSetInput = (_e: any, _stateVar: any, _stateVarName: string) => {
+		let isNum, isFloat, isUglyFloat1, isUglyFloat2, clampDecimals;
+
+		_stateVar = _e.currentTarget.value;
+		isNum = /^\d+$/.test(_stateVar);
+		isFloat = /^[0-9]+\.[0-9]+$/.test(_stateVar);
+		isUglyFloat1 = /^\.[0-9]+$/.test(_stateVar);
+		isUglyFloat2 = /^[0-9]+\.$/.test(_stateVar);
+		clampDecimals = /^\d+\.\d{0,3}$/.test(_stateVar);
+
+		if (isNum || isFloat || isUglyFloat1 || isUglyFloat2 || clampDecimals) {
+			if (_stateVar.indexOf(' ') >= 0) {
+				// if includes whitespace
+				_stateVar.trim();
+			}
+
+			if (!isNaN(_stateVar) && _stateVar !== '') {
+				if (_stateVarName === 'entryPrice') {
+					const clampedInput = parseFloat(parseFloat(_stateVar).toFixed(2));
+					setEntryPrice(ethers.BigNumber.from(clampedInput * scalar));
```

Comment:
Sticking to `BigNumber` or `Wei` types throughout will avoid all of these `parseFloat` hacks